task/fluxlocal/query.go

```diff
@@ -53,17 +54,31 @@ func (f *fluxQueryer) injectDependencies(ctx context.Context) context.Context {
 	}
 
 	// inject the dependencies to the context.
-	return deps.Inject(ctx)
+	return dependency.Inject(ctx, deps)
 }
 
 func (f *fluxQueryer) Query(ctx context.Context, compiler flux.Compiler) (flux.ResultIterator, error) {
 	f.logger.Info("executed flux query")
-	ctx = f.injectDependencies(context.Background())
+	var span *dependency.Span
+	ctx, span = f.injectDependencies(context.Background())
```

Comment:
Why use a new `context.Background` when a `context.Context` is passed in?  Should that be used instead of being shadowed?